server/src/main/java/com/google/speech/tools/voxetta/services/DatastorePromptService.java

```diff
@@ -0,0 +1,132 @@
+
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+
+
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+
+import com.google.speech.tools.voxetta.data.Prompt;
+import com.google.speech.tools.voxetta.data.PromptBuilder;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Implements prompt service using Google Datastore API.
+ */
+public class DatastorePromptService implements PromptService {
+
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private Gson gson = new Gson();
+
+    public DatastorePromptService() {
+    }
+
+    @Override
+    public boolean savePrompt(String type, String body) {
+
+        // TODO: validation
+
+        Entity promptEntity = new Entity("Prompt");
+
+        promptEntity.setProperty("type", type);
+        promptEntity.setProperty("body", body);
+
+        promptEntity.setProperty("read", 0);
+
+        datastore.put(promptEntity);
+
+        return true;
+    }
+
+    /**
+     * Returns one prompt from the database.
+     * <p>
+     * When it is retrieved, its "read" property is updated to 1. Once that happens, it can no
+     * longer be retrieved via this function.
+     *
+     * @return one prompt from the Prompt database.
+     */
+    @Override
+    public String getOnePrompt() {
+
+        // get the prompt
+        Filter unreadFilter = new FilterPredicate("read", FilterOperator.EQUAL, 0);
+        List<Entity> unreadQueries = datastore.prepare(new Query("Prompt")
+            .setFilter(unreadFilter))
+            .asList(Builder.withLimit(1));
+
+        // if none, return empty json
+        if (unreadQueries.size() < 1) {
+            return gson.toJson(unreadQueries);
+        }
+
+        Entity retrievedEntity = unreadQueries.get(0);
+
+        // set prompt as read
+        retrievedEntity.setProperty("read", 1);
+        datastore.put(retrievedEntity);
+
+        // return the prompt as JSON
+        Prompt retrievedPrompt = new PromptBuilder().buildFromEntity(retrievedEntity);
+        return gson.toJson(retrievedPrompt);
+    }
+
+    /**
+     * Resets the read status of all prompts to 0 so they can be used again. Primarily for testing.
+     *
+     * @return a boolean denoting success or failure.
+     */
+    public boolean resetAllToUnread() {
+
+        Iterable<Entity> iterableResults = datastore.prepare(new Query("Prompt")).asIterable();
+
+        for (Entity entity : iterableResults) {
+            entity.setProperty("read", 0);
+            datastore.put(entity);
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns all prompts as entities. Primarily for testing.
+     *
+     * @return JSON of all prompts as entities.
+     */
+    public String getAllPrompts() {
```

Comment:
`@VisibleForTesting` Or put in a test utils class

Reply:
this isn't for automated testing but more for me to see what's going on in postman, and because of that I'm not really sure if I should tag this for testing